products/views.py

```diff
@@ -1,3 +1,27 @@
-from django.shortcuts import render
+from django.http     import JsonResponse
+from django.views    import View
 
-# Create your views here.
+from products.models import Category
+
+
+class CategoryView(View):
+    def get(self, request):
+        category_data = [
+            {
+                'category_name' : category.name,
+                'ml_volume'     : category.ml_volume,
+                'price'         : category.price,
+                'products'      : [
+                            {
+                                'thumb' : product.image_set.get(name='thumb').url,
+                                'tags'  : [i.name for i in product.tags.all()],
+                                'id'    : product.id
+                            }
+                for product in category.product_set.all()]
+            }
+            for category in Category.objects.all()
+        ]
+
+        return JsonResponse({'MESSAGE': 'SUCCESS', 'RESULT': category_data}, status=200) 
```

Comment:
깔끔하게 잘 구현해 주셨네요!
1. 카테고리 목록을 반환하는 경우에는 class명을 복수로 만들어주세요~
2. 변수명을 정할 때 data라는 용어는 제외해도 좋겠습니다.

```suggestion
class CategoriesView(View):
    def get(self, request):
        categories = [{
            'name' : category.name,
            'ml_volume'     : category.ml_volume,
            'price'         : category.price,
            'products'      : [{
                'thumb' : product.image_set.get(name='thumb').url,
                'tags'  : [i.name for i in product.tags.all()],
                'id'    : product.id
            } for product in category.product_set.all()]
        } for category in Category.objects.all()]

        return JsonResponse({'MESSAGE': 'SUCCESS', 'RESULT': categories}, status=200) 
```

Reply:
수정했습니당!